apps/hvn/src/components/PayloadForm/PayloadForm.tsx

```diff
@@ -5,32 +5,36 @@ import { SubmitHandler, useForm } from 'react-hook-form'
 
 import { Button } from '@fc/ui/base/button'
 import { Card, CardContent, CardDescription, CardFooter, CardHeader, CardTitle } from '@fc/ui/base/card'
-import { Checkbox } from '@fc/ui/base/checkbox'
-import { Input } from '@fc/ui/base/input'
-import { Label } from '@fc/ui/base/label'
-import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@fc/ui/base/select'
-import { Textarea } from '@fc/ui/base/textarea'
+import { FormCheckbox } from '@fc/ui/components/form/form-checkbox'
+import { FormInput } from '@fc/ui/components/form/form-input'
+import { FormSelect } from '@fc/ui/components/form/form-select'
+import { FormTextarea } from '@fc/ui/components/form/form-textarea'
 
 import { camelCase } from 'lodash'
-import { LinkIcon, Send } from 'lucide-react'
+import { LinkIcon, Loader2, Send } from 'lucide-react'
 import { useLocale, useTranslations } from 'next-intl'
 import Link from 'next/link'
 import { toast } from 'sonner'
 
-import type { Form } from '../../../payload-types'
-interface FormProps {
-  formData: Form
+import { Form } from '@fc/ui/base/form'
+import type { Form as FormData } from '../../../payload-types'
+
+interface PayloadFormProps {
+  formData: FormData
 }
 
-export const PayloadForm: React.FC<FormProps> = ({ formData }) => {
-  const { handleSubmit, register, reset } = useForm<Record<string, string>>()
+export const PayloadForm: React.FC<PayloadFormProps> = ({ formData }) => {
+  const form = useForm<Record<string, string>>()
```

Comment:
We miss frontend validation with this implementation. That's why I was considering to make custom frontend form that is not related to backend form data.

We could have a function that generates zod schema based on the payload form data but I wouldn't put so much effort on this. Because we don't have many forms and it doesn't have to be super dynamic.

We can build a fully dynamic form in the future when we decide to move dashboard app to payload.